datajoint/external.py

```diff
@@ -304,27 +304,37 @@ def download_filepath(self, filepath_hash):
         :param filepath_hash: The hash (UUID) of the relative_path
         :return: hash (UUID) of the contents of the downloaded file or Nones
         """
+
+        def _need_checksum(local_filepath):
+            limit = config.get("filepath_checksum_size_limit")
+            return limit is None or Path(local_filepath).stat().st_size < limit
+
         if filepath_hash is not None:
             relative_filepath, contents_hash = (self & {"hash": filepath_hash}).fetch1(
                 "filepath", "contents_hash"
             )
             external_path = self._make_external_filepath(relative_filepath)
             local_filepath = Path(self.spec["stage"]).absolute() / relative_filepath
-            file_exists = (
-                Path(local_filepath).is_file()
-                and uuid_from_file(local_filepath) == contents_hash
+
+            file_exists = Path(local_filepath).is_file() and (
+                not _need_checksum(local_filepath)
+                or uuid_from_file(local_filepath) == contents_hash
             )
+
             if not file_exists:
                 self._download_file(external_path, local_filepath)
-                checksum = uuid_from_file(local_filepath)
-                if (
-                    checksum != contents_hash
-                ):  # this should never happen without outside interference
-                    raise DataJointError(
-                        "'{file}' downloaded but did not pass checksum'".format(
-                            file=local_filepath
+                if _need_checksum(local_filepath):
+                    checksum = uuid_from_file(local_filepath)
+                    if checksum != contents_hash:
+                        # this should never happen without outside interference
+                        raise DataJointError(
+                            "'{file}' downloaded but did not pass checksum'".format(
+                                file=local_filepath
+                            )
```

Comment:
```suggestion
                    uuid_from_file(local_filepath) != contents_hash:
                        # this should never happen without outside interference
                        raise DataJointError(f"'{local_filepath}' downloaded but did not pass checksum'")
```

Reply:
I have changed this `.format` to an fstring. I also changed a few more in external.py to fstrings but not all of them, some of them are sql queries which were too difficult to change in a short pass. If you want I can change those as well but it will take some time.